pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -171,39 +200,101 @@ var _ = Describe("NodeImageProvider tests", func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
+		})
+
+		//TODO: Modify when Ubuntu 22.04 with FIPS becomes available
+		It("should match expected images for FIPS Ubuntu2204 when FIPSMode is explicitly set to FIPS", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.Ubuntu2204ImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr("FIPS")
+
+			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundImages).To(BeEmpty())
+		})
+
+		It("should match expected images for FIPS Ubuntu2004 when FIPSMode is explicitly set to FIPS", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.UbuntuImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr("FIPS")
+
+			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundImages).To(ContainElements(getExpectedTestFIPSImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
+		})
+
 		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
 		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
-		It("should match expected images for AzureLinux, depending on the Kubernetes version", func() {
+		It("should match expected images for default AzureLinux, depending on the Kubernetes version, when FIPSMode is explicitly set to Disabled", func() {
 			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr("Disabled")
 
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
 		})
 
-		It("should match expected images for AzureLinux with version < 1.32", func() {
+		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
+		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
+		It("should match expected images for default AzureLinux, depending on the Kubernetes version, when FIPSMode is not explicitly set", func() {
 			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
-			nodeClass.Status.KubernetesVersion = "1.31.0"
 
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, "1.31.0")))
+			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
 		})
 
-		It("should match expected images for AzureLinux with version >= 1.32", func() {
```

Comment:
Rather than modifying these tests, I would add new tests in addition to the existing ones.

Goes for the existing:
- `"should match expected images for Ubuntu2204"`
- `"should match expected images for AzureLinux, depending on the Kubernetes version"`
- `"should match expected images for AzureLinux with version < 1.32"`